apps/mobile/src/components/GalleryEditor/GalleryEditorRow.tsx

```diff
@@ -0,0 +1,97 @@
+import clsx from 'clsx';
+import React, { useCallback } from 'react';
+import { GestureResponderEvent, useWindowDimensions, View, ViewProps } from 'react-native';
+import Animated from 'react-native-reanimated';
+import { graphql, useFragment } from 'react-relay';
+
+import { useGalleryEditorActions } from '~/contexts/GalleryEditor/GalleryEditorContext';
+import { StagedRow } from '~/contexts/GalleryEditor/types';
+import { GalleryEditorRowFragment$key } from '~/generated/GalleryEditorRowFragment.graphql';
+
+import { GalleryTouchableOpacity } from '../GalleryTouchableOpacity';
+import { GalleryEditorActiveActions } from './GalleryEditorActiveActions';
+import { GalleryEditorTokenPreview } from './GalleryEditorTokenPreview';
+import { horizontalRowPadding, inBetweenColumnPadding } from './utils';
+
+type Props = {
+  sectionId: string;
+  row: StagedRow;
+  style?: ViewProps['style'];
+  queryRef: GalleryEditorRowFragment$key;
+};
+
+export function GalleryEditorRow({ sectionId, row, style, queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment GalleryEditorRowFragment on Query {
+        ...GalleryEditorActiveActionsFragment
+      }
+    `,
+    queryRef
+  );
+
+  const { activateRow, activeRowId } = useGalleryEditorActions();
+
+  const screenDimensions = useWindowDimensions();
+
+  const column = row.columns;
+  const totalSpaceForTokens =
+    screenDimensions.width - horizontalRowPadding * 2 - inBetweenColumnPadding * (column - 1);
+
+  const widthPerToken = totalSpaceForTokens / column;
```

Comment:
nit:
- `column` -> `columns`
- let's memoize `widthPerToken`. in fact we can probably make this a helper hook, like `useWidthPerToken(columns: number)`

Reply:
Nice helper! Updated it to use `useWidthPerToken`